application.py

```diff
@@ -1,13 +1,478 @@
+        )
+    except Exception as error:
+        logger.error(error)
+        flash("Please re-login to authenticate your account.")
+        return redirect(url_for("home"))
+
+
+@app.route("/project/<project_name>", methods=["GET"])
+def view_project(project_name):
+    headers = {"Authorization": f"{session['id_token']}"}
+    projects = requests.get(
+        f"{API_URL}/api/projects/{project_name}",
+        headers=headers,
+    ).json()
+    # projects either returnes {'description': 'Project not found', 'message': None} or a project in dict form.
+    print(projects)
+    try:
+        if projects["message"] and projects["message"] is None:
+            flash("Project not found. Please try again.")
+            return redirect(url_for("dashboard"))
+    except Exception:
+        return render_template("view_project.html", project=projects)
+
+
+@app.route("/survey", methods=["GET", "POST"])
+def survey():
+    # ONLY GET AND POST ARE ALLOWED HENCE NO NEED FOR SECOND IF STATEMENT
+    # IF METHOD IS 'GET' THEN THE SURVEY IS RENDERED
+    if request.method == "GET":
+        return render_template("survey.html")
+    # IF METHOD IS 'NOT GET' THEN THE POST PROCESS BEGINS
+    headers = {
+        "Authorization": f"{session['id_token']}",
+        "content-type": "application/json",
+    }
+
+    user = json.loads(request.form["user"])
+    project = json.loads(request.form["project"])
+    developed = json.loads(request.form["developed"])
+    source_control = json.loads(request.form["source_control"])
+    hosting = json.loads(request.form["hosting"])
+    database = json.loads(request.form["database"])
+    languages = json.loads(request.form["languages"])
+    frameworks = json.loads(request.form["frameworks"])
+    integrations = json.loads(request.form["integrations"])
+    infrastructure = json.loads(request.form["infrastructure"])
+    stage = json.loads(request.form["stage"])
+
+    developed_company = ""
+    if developed["developed"] == "Outsourced":
+        developed_company = developed["outsource_company"]
+    elif developed["developed"] == "Partnership":
+        developed_company = developed["partnership_company"]
+
+    data = {
+        "user": [u for u in user],
+        "details": [
+            {
+                "name": project["project_name"],
+                "short_name": project["project_short_name"],
+                "documentation_link": [project["doc_link"]],
+                "project_description": project["project_description"],
+            }
+        ],
+        "developed": [developed["developed"], [developed_company]],
+        "source_control": source_control,
+        "architecture": {
+            "hosting": hosting,
+            "database": database,
+            "languages": languages,
+            "frameworks": frameworks,
+            "CICD": integrations,
+            "infrastructure": infrastructure,
+        },
+        "stage": stage,
+    }
+    print(data)
```

Comment:
Can we use logging here rather than print.

Reply:
Removed this print. Changed the other lines that printed errors into logger.error().